### PR TITLE
Use spf13/cobra for docker search

### PR DESCRIPTION
**Pull request by vdemeester**
- Move image search command search to `api/client/image/search.go`
- Use cobra 🐍.

/cc @dnephin @thaJeztah @LK4D4 

🐸

Signed-off-by: Vincent Demeester vincent@sbr.pm

### CODE REVIEW

**Comment by vdemeester** [comment]
@dnelphin there is one weird thing though, putting `[OPTIONS] …` into `Use` make the output of `help` weird  :

```bash
λ ./bundles/latest/dynbinary-client/docker help
# […]
    save      Save one or more images to a tar archive
    search [OPSearch the Docker Hub for images
    start     Start one or more stopped containers
# […]
```

but the deprecated feature, I :heart: it : 

```bash
λ ./bundles/latest/dynbinary-client/docker search --automated docker
Flag --automated has been deprecated, Use --filter=automated=true instead
NAME                                    DESCRIPTION                                     STARS     OFFICIAL   AUTOMATED
konradkleine/docker-registry-frontend   Browse and modify your Docker registry in ...   95                   [OK]
jdubois/jhipster-docker                 DO NOT USE: Former official JHipster Docke...   57                   [OK]
martin/docker-cleanup-volumes           Delete orphaned docker volumes                  47                   [OK]
datadog/docker-dd-agent                 Docker container for the Datadog Agent.         28                   [OK]
```

**Comment by dnephin** [comment]
@vdemeester I fixed that here: https://github.com/docker/docker/pull/23159/files#diff-4a2b6f090171fe18f7406cb69fb2e479R54 so it should be fine once that PR is merged.  Doing the rebase now.

**Comment by vdemeester** [comment]
@dnelphin perfect ! :+1:

**Comment by dnephin** [comment]
LGTM

File: cli/required.go [hunk 1]
```diff
@@ -40,3 +40,19 @@ func RequiresMinArgs(min int) cobra.PositionalArgs {
...
 }
+
+// ExactArgs returns an error if there is not the exact number of args
+func ExactArgs(number int) cobra.PositionalArgs {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) == number {
+			return nil
+		}
+		return fmt.Errorf(
+			"\"%s\" requires exactly %d argument(s).\n\nUsage:  %s\n\n%s",
```
Comment (thaJeztah) [inline cli/required.go]: nit: this is different from what we have now; would be nice if we have a generic function to use singular/plural here

```
docker: "search" requires 1 argument.
```

**Comment by thaJeztah** [comment]
Funny; it treats `--limit=-1` and  `--limit -1` different;
```
docker search --limit=-1 a
Error response from daemon: Limit -1 is outside the range of [1, 100]

docker search --limit -1 a
"docker search" requires exactly 1 argument(s).

Usage:  docker search [OPTIONS] TERM
...
```

LGTM

**Comment by cpuguy83** [comment]
LGTM, windows error is know, gcc finished but didn't notify.

**Pull request merged**